src/engine/init.c

```diff
@@ -628,6 +628,9 @@ static int
 server_init(int argc, char *argv[])
 {
 	uint64_t		bound;
+	uint64_t		metrics_size;
+	const uint64_t		est_std_metrics = 1024; /* high estimate to allow for pool links */
+	const uint64_t		est_tgt_metrics = 128; /* high estimate */
```

Comment:
[Not requesting changes] Would be better to put these into a new function that takes a number of targets and returns a size. Because `server_init` involves many different "things," we should avoid introducing several variables for a single "thing."

Reply:
That is good advice, thank you. If I get any comments requiring me to re-push, I'll make this change.

Reply:
Thanks. :)

Reply:
For what it's worth, I also think that it would be a good idea to encapsulate this logic in a `d_tm_` helper. Currently we only use the library on the server, but there are loose plans to use it on the client side as well (e.g. [DAOS-8851](https://daosio.atlassian.net/browse/DAOS-8851)).

Reply:
I felt like this didn't belong in gurt as a d_tm_* function because it contains knowledge of the actual usage of the metrics in the server. (Pools as links to other regions, targets, etc.) I think it makes a good static helper function within this file though.

The calculation for clients would be a little different anyway, I think, since the metric set would be different.

Reply:
Got some errors in tests, so I went ahead and made the change @liw suggested.